### PR TITLE
chore: upgrade envoy to newest tag, and use patch to export symbols

### PR DESCRIPTION
**Pull request by rainingmaster**

### CODE REVIEW

File: envoy-golang.yaml [hunk 1]
```diff
@@ -5,7 +5,7 @@ bootstrap_extensions:
   typed_config:
     "@type": type.googleapis.com/envoy.extensions.dso.v3.dso
     so_id: mosn
-    so_path: "/usr/local/envoy-go-extension/libgolang.so"
+    so_path: "/root/github.com/envoy-go-extension/libgolang.so"
```
Comment (doujiang24) [inline envoy-golang.yaml:8]: better not edit this, it's used for building docker image.

File: envoy.bazelrc [hunk 2]
```diff
@@ -38,10 +37,10 @@ common --experimental_allow_tags_propagation
 # (Workaround for https://github.com/bazelbuild/rules_foreign_cc/issues/421)
 build:linux --copt=-fPIC
 build:linux --copt=-Wno-deprecated-declarations
-build:linux --cxxopt=-std=c++17
+build:linux --cxxopt=-std=c++17 --host_cxxopt=-std=c++17
 build:linux --conlyopt=-fexceptions
 build:linux --fission=dbg,opt
-# build:linux --features=per_object_debug_info
+build:linux --features=per_object_debug_info
```
Comment (doujiang24) [inline envoy.bazelrc:43]: better keep this commentted. this feature may cause bug while we are debug using gdb.

**Review by doujiang24 — COMMENTED** [review]
@rainingmaster Nice job. just a few comments.

File: envoy.bazelrc (hunk 2 above)
Comment (doujiang24) [reply envoy.bazelrc:43]: bug means gdb may failed to read seperate dwp file, which is enabled by this this feature.

**Comment by rainingmaster** [comment]
@doujiang24 all are done, please help to review again, thx!

**Pull request merged**